opentracing_shim/src/main/java/io/opentelemetry/opentracingshim/SpanBuilderShim.java

```diff
@@ -192,11 +195,15 @@ public Span start() {
     if (ignoreActiveSpan && parentSpan == null && parentSpanContext == null) {
       builder.setNoParent();
     } else if (parentSpan != null) {
-      builder.setParent(parentSpan.getSpan());
+      // Note: We ignore the (potentially stored) parentSpan's (grand)parent context here.
+      builder.setParent(TracingContextUtils.withSpan(parentSpan.getSpan(), Context.current()));
       SpanContextShim contextShim = spanContextTable().get(parentSpan);
       distContext = contextShim == null ? null : contextShim.getCorrelationContext();
     } else if (parentSpanContext != null) {
-      builder.setParent(parentSpanContext.getSpanContext());
+      // TODO: This might be wonky
```

Comment:
can you explain the TODO here? Do we need an issue to track a change? I'd prefer not to leave a cryptic TODO in the code like this, since it's not clear what the issue is.

Reply:
I would need @carlosalberto to check this. It just felt a bit strange, but I don't have enough knowledge of the OpenTracing shim to verify if there is a problem.

Reply:
Maybe SpanContextShim should store a Context.

Reply:
I created #1699 and will change this to use ROOT instead of current (that way later changes are less likely to be breaking).

Reply:
Done in 9f618e0